Technomancy/app.ts

```diff
@@ -3835,16 +3835,19 @@ let autobuild_amount = 0;
 let autobuild_repeat = false;
 function draw_autobuild() {
     let autobuild_slots = 10;
-    if (adventure_data["tower_floor"] > 25) {
+    if ((adventure_data["tower_floor"] > 25) || adventure_data["tower_ascension"]) {
         autobuild_slots += 10;
     }
-    if (adventure_data["tower_floor"] > 26) {
-        autobuild_slots += adventure_data["tower_floor"];
+    if ((adventure_data["tower_floor"] > 26) || adventure_data["tower_ascension"]) {
+        if ((adventure_data["tower_ascension"]) && (adventure_data["tower_floor"] < TOWER_DATA.length - 2 + (adventure_data["tower_ascension"] - 1) * tower_ascension_growth)) {
```

Comment:
Hmm, might be good to refactor `TOWER_DATA.length - 2 + (adventure_data["tower_ascension"]) * tower_ascension_growth)` into some function like `tower_height()` and then use `tower_height() - tower_ascension_growth`. Should just save a bit of pain when re-reading that code. (There's a few other places where you can use tower_height too, which is why I'm mentioning this). I'd accept without this change, but then probably make it myself at some point.

Reply:
Done, for other reasons.
Now you can have additional hardcoded bosses beyond ascension as ascension point does not depend on length of TOWER_DATA.

Reply:
Excellent. This should work fine now. (Well, the 30 + X might be off by one, I'd have to test it)